### ISSUE TITLE
Mocking mode still attempts to create a socket

### ISSUE THREAD

**Issue by runk** [brightcove/hot-shots#92]
Problem: even with `{ mock: true }` option enabled, new socket is created, and connection attempted. It raises unwanted noise in errors due that.

Ref: https://github.com/brightcove/hot-shots/blob/master/lib/statsd.js#L68

**Comment by bdeitte** [comment]
Makes sense to fix, and happy to take PRs from anyone on it.  Looks like there would also need to be some this.socket cleanup in there, of making sure there is a valid reference before it is used in places.

**Comment by TheMagoo73** [comment]
Happy to take a run at this one.

Having had a quick look at this, I'm wondering if an option would be to create a mock transport that the `createTransport` sets up if the mock flag is set. I think that would be a less invasive change that's simpler to test, and could be extended in future, e.g. to dump the stats to the console.

**Comment by bdeitte** [comment]
@TheMagoo73 I think that's a great idea, thanks!
